### PR TITLE
Implement core Graph model + TEST

### LINKED ISSUE #5: Implement core Graph model + TEST
Implement a modular Python graph model with classes Node, Edge, and Graph, supporting directed/undirected and cyclic/acyclic graphs. 
All classes should include type validation, get/set methods for attributes, and serialization/deserialization. 
The Graph must support adding/removing nodes and edges, retrieving neighbors, and detecting cycles. 
Write unit tests covering attribute validation, graph operations, and serialization.

### PR DESCRIPTION
**Pull request by paunovicbojana**
Hi Team,

I’ve implemented the core graph data model, including Node, Edge, and Graph classes with support for directed/undirected and cyclic/acyclic graphs. Attribute validation has been added to ensure type consistency (int, float, str, date). I also included adjacency lists for efficient neighbor and edge lookups, along with serialization/deserialization methods. Unit tests for nodes, edges, and graph operations are added to cover basic functionality and edge cases. 

Please let me know if you have any feedback or suggestions for improvement.

Closes #5

### CODE REVIEW

File: api/src/tangled_api/model.py [hunk 1]
```diff
@@ -69,28 +238,272 @@ def __init__(self, directed: bool = True):
...
+        visited = set()
+        
+        def dfs(node_id, parent_id):
+            visited.add(node_id)
+            
+            for neighbor in self.get_neighbors(node_id):
+                if neighbor.id not in visited:
+                    if dfs(neighbor.id, node_id):
+                        return True
+                elif neighbor.id != parent_id:
```
[the remark this answers is not in the document]
Comment (Vukotije) [reply api/src/tangled_api/model.py:457]: Don't know if we should support having multiple edges between two nodes?

**Review by re-pixel — APPROVED** [review]
Graph model is now compatible with the rest of Tangled project
